sdk/include/reactos/dbgbitmap.h

```diff
@@ -0,0 +1,359 @@
+RtlClearBitDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _In_range_(<, BitMapHeader->SizeOfBitMap) ULONG BitNumber)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    if (RtlCheckBit(&BitMapHeader->BitMap, BitNumber) == FALSE)
+    {
+        BitMapHeader->NumberOfSetBits--;
+    }
+    RtlClearBit(&BitMapHeader->BitMap, BitNumber);
+    BitMapHeader->BitmapHash = RtlComputeBitmapHashDbg(BitMapHeader);
+}
+
+static inline
+VOID
+RtlClearBitsDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _In_range_(0, BitMapHeader->SizeOfBitMap - NumberToClear) ULONG StartingIndex,
+    _In_range_(0, BitMapHeader->SizeOfBitMap - StartingIndex) ULONG NumberToClear)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    RtlClearBits(&BitMapHeader->BitMap, StartingIndex, NumberToClear);
+    BitMapHeader->NumberOfSetBits = RtlNumberOfSetBits(&BitMapHeader->BitMap);
+    BitMapHeader->BitmapHash = RtlComputeBitmapHashDbg(BitMapHeader);
+}
+
+static inline
+ULONG
+RtlFindClearBitsDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _In_ ULONG NumberToFind,
+    _In_ ULONG HintIndex)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindClearBits(&BitMapHeader->BitMap, NumberToFind, HintIndex);
+}
+
+static inline
+ULONG
+RtlFindClearBitsAndSetDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _In_ ULONG NumberToFind,
+    _In_ ULONG HintIndex)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindClearBitsAndSet(&BitMapHeader->BitMap, NumberToFind, HintIndex);
+}
+
+static inline
+ULONG
+RtlFindFirstRunClearDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _Out_ PULONG StartingIndex)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindFirstRunClear(&BitMapHeader->BitMap, StartingIndex);
+}
+
+static inline
+ULONG
+RtlFindClearRunsDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _Out_writes_to_(SizeOfRunArray, return) PRTL_BITMAP_RUN RunArray,
+    _In_range_(>, 0) ULONG SizeOfRunArray,
+    _In_ BOOLEAN LocateLongestRuns)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindClearRuns(&BitMapHeader->BitMap, RunArray, SizeOfRunArray, LocateLongestRuns);
+}
+
+static inline
+ULONG
+RtlFindLastBackwardRunClearDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _In_ ULONG FromIndex,
+    _Out_ PULONG StartingRunIndex)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindLastBackwardRunClear(&BitMapHeader->BitMap, FromIndex, StartingRunIndex);
+}
+
+static inline
+ULONG
+RtlFindLongestRunClearDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _Out_ PULONG StartingIndex)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindLongestRunClear(&BitMapHeader->BitMap, StartingIndex);
+}
+
+static inline
+ULONG
+RtlFindNextForwardRunClearDbg(
+    _In_ PRTL_BITMAP_DBG BitMapHeader,
+    _In_ ULONG FromIndex,
+    _Out_ PULONG StartingRunIndex)
+{
+    RtlValidateBitmapDbg(BitMapHeader);
+    return RtlFindNextForwardRunClear(&BitMapHeader->BitMap, FromIndex, StartingRunIndex);
+}
+
+NTSYSAPI
+ULONG
+NTAPI
+RtlFindNextForwardRunSet(
```

Comment:
We don't have this one in our headers?

Reply:
It's in NDK, but not in wdm.h (Windows DDK also doesn't have it). I can add it to wdm.h instead, if that is preferred. Would make sense imo.

Reply:
Done

Reply:
Is this a windows 8+ export of ntoskrnl or sthg? I don't see it (I only see the functions for the largest cleared set)

Reply:
I don't know. I just cannot figure out, how to figure that out. There is no documentation, nothing in our codebase, and no tools to figure that out otherwise. So I guess we will all be doomed to never know.